### PR TITLE
feat(@angular-ru/common): add partition array method

### PR DESCRIPTION
**Pull request by Schirbak**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by mansurishmuhametov — APPROVED** [review]

File: packages/common/array/src/partition.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Couple } from '@angular-ru/common/typings';
+
+export function partition<T>(
+    forArray: T[],
+    callback: (element: T, index?: number, array?: T[]) => boolean
```
Comment (splincode) [inline packages/common/array/src/partition.ts:5]: `callback` -> `predicate`

File: packages/common/array/src/partition.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+): Couple<T[]> {
+    return forArray.reduce(
+        (partitioning: Couple<T[]>, element: T, ...rest: [number, T[]]): Couple<T[]> => {
+            partitioning[callback(element, ...rest) ? 0 : 1].push(element);
```
Comment (splincode) [inline packages/common/array/src/partition.ts:9]:
```
callback(element, ...rest) ? 0 : 1
```
лучше вынести в переменную и написать семантику этого индекса (в виде jsdoc)

Comment (Schirbak) [reply packages/common/array/src/partition.ts:9]: 👌

File: packages/common/array/src/partition.ts (hunk 1 above)
Comment (Schirbak) [reply packages/common/array/src/partition.ts:5]: почему интересно я его назвал callback 🤔

**Review by splincode — APPROVED** [review]

**Pull request merged**